opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -175,6 +178,24 @@ def __init__(
 
         self._shutdown = False
 
+        self._views = views
```

Comment:
Sorry, where are they needed?

Reply:
when creating the ViewStorage/ViewInstrumentMatch objects

Reply:
Hm, not sure we need them to create those objects, at least in [this](https://github.com/open-telemetry/opentelemetry-python/pull/2301/files#diff-693118fd4a59966d0e9d2499bb9704ce0af972e3390e712b3a7488e0848009bbR76-R95) way we don't.

Reply:
> Hm, not sure we need them to create those objects, at least in [this](https://github.com/open-telemetry/opentelemetry-python/pull/2301/files#diff-693118fd4a59966d0e9d2499bb9704ce0af972e3390e712b3a7488e0848009bbR76-R95) way we don't.

```py
                    # pylint: disable=protected-access
                    for view in self._meter_provider._views
```

the `sdk_config` is passed around to accomplish the above without doing private access. See [here](https://github.com/open-telemetry/opentelemetry-python/blob/fd9862aee1ef305fca39879931c92429d57106d6/opentelemetry-sdk/src/opentelemetry/sdk/_metrics/measurement_consumer.py#L54)

Reply:
Sure we can do that, but actually, aren't the `resource`, `metric_readers` (and with this change `views`) attributes of the `MeterProvider` the same as `SdkConfiguration`? The `XMeasurementConsumer` receives an `SdkConfiguration` object, we could as well as make the `MeterProvider` `metric_readers`, `resource` and `views` attributes public (they are objects that are passed to the `MeterProvider` constructor, anyways) and just pass the `MeterProvider` to the `XMeasurementConsumer` instead of having `SdkConfiguration`. :shrug:

Reply:
Making these fields public implies that they can be configured during runtime which I believe should not be allowed.